File: ArbitageCalculator.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>

using namespace std;

int main(int argc, char const *argv[])
{
	ifstream file1("sportsbet_Odds.csv");
	vector<string> names;

	string game;
	string name1;
	string over_Odd1;
	string under_Odd1;
	string point1;

	vector<string> names1;
	vector<double> over_Odds1;
	vector<double> under_Odds1;
	vector<double> points1;

	// Extracting odds from sportsbet_Odds.csv
	while(file1.good())
	{
		getline(file1,name1,',');
		getline(file1,over_Odd1,',');
		getline(file1,under_Odd1,',');
		getline(file1,point1,'\n');

		names1.push_back(name1);
		names.push_back(name1);
		over_Odds1.push_back(stod(over_Odd1));
		under_Odds1.push_back(stod(under_Odd1));

		point1 = point1.substr(point1.find("+")+1,point1.find(")")-point1.find("+"));
		points1.push_back(stod(point1));
	}


	// cout << "******Sportsbet Odds******" << endl;
	for (int i = 0; i < names1.size()-1; i++)
	{
		names1[i] = names1[i].erase(names1[i].find("-")-1,names1[i].find("|")-(names1[i].find("-")-1));
		// cout << names1[i] << endl;
	}

	ifstream file2("bet365_Odds.csv");

	getline(file2,game,'\n');

	string name2;
	string over_Odd2;
	string under_Odd2;
	string point2;

	vector<string> names2;
	vector<double> over_Odds2;
	vector<double> under_Odds2;
	vector<double> points2;

	// Extracting odds from bet365_Odds.csv
	while(file2.good())
	{
		getline(file2,name2,',');
		getline(file2,over_Odd2,',');
		getline(file2,under_Odd2,',');
		getline(file2,point2,'\n');

		names2.push_back(name2);
		over_Odds2.push_back(stod(over_Odd2));
		under_Odds2.push_back(stod(under_Odd2));

		points2.push_back(stod(point2));
	}

	double bestOver = 0.0;
	double bestUnder = 0.0;

	// Looking for matching games
	for (int i = 0; i < names1.size()-1; i++)
	{
		for (int j = 0; j < names2.size()-1; j++)
		{
			if (names1[i]==names2[j])
			{
				cout << "################################" << endl; 
				cout << game << endl << endl;
				cout << "Same Game Found: " << names[i] << endl;

				// When games have different points we are not actually betting on the same thing
				// So we skip these.
				if (points1[i]!=points2[j])
				{
					cout << "Points are different :(" << endl << endl;

					cout << "Sportsbet Points " << points1[i] << endl;
					cout << "Bet365 Points " << points2[j] << endl;
					cout << "################################" << endl << endl; 
					continue;
				}

				if(over_Odds1[i]>over_Odds2[j])
				{
					bestOver = over_Odds1[i];
					bestUnder = under_Odds2[j];
				} else {
					bestOver = over_Odds2[j];
					bestUnder = under_Odds1[i];
				}

				//arbitage formula -> https://youtu.be/TGinzvSDayU?t=333 
				const auto calc = (1/bestOver+1/bestUnder)*100;

				if (calc < 100)
				{
					cout << "ARBITAGE FOUND!!! -> " << names[i] << endl << endl; 
					cout << "Profit margin of -> " << 100-calc << endl << endl; 

					cout << "Sportsbet Over " << over_Odds1[i] << endl;
					cout << "Sportsbet Under " << under_Odds1[i] << endl << endl;

					cout << "Bet365 Over " << over_Odds2[j] << endl;
					cout << "Bet365 Under " << under_Odds2[j] << endl << endl;

					cout << "Best Over " << bestOver << endl; 
					cout << "Best Under " << bestUnder << endl ; 
				} else {
					cout << "not an arbitage :(" << endl;
				}
				cout << "################################" << endl << endl; 
			}
		}
	}


	return 0;
}

File: ArbitageCalculator.py
import csv
budget = 800
class ArbitageCalculator:
    def lets_go_csv(self):
        sportsbet_names = []
        sportsbet_overOdds = []
        sportsbet_underOdds = []
        sportsbet_points = []

        with open('sportsbet_Odds.csv') as csv_file :

            csv_reader = csv.reader(csv_file, delimiter=',')
            for row in csv_reader:
                sportsbet_names.append(row[0][:row[0].find("-")-1] + row[0][row[0].find("|"):])
                sportsbet_overOdds.append(row[1])
                sportsbet_underOdds.append(row[2])
                sportsbet_points.append(row[3][row[3].find("+")+1:-1])

        bet365_names = []
        bet365_overOdds = []
        bet365_underOdds = []
        bet365_points = []

        with open('bet365_Odds.csv') as csv_file :

            csv_reader = csv.reader(csv_file, delimiter=',')
            for row in csv_reader:
                # if(len(row)==1):
                #     game_name = row[0] 
                #     continue
                bet365_names.append(row[0])
                bet365_overOdds.append(row[1])
                bet365_underOdds.append(row[2])
                bet365_points.append(row[3])


        best_over = 0.0;
        best_under = 0.0;
        games_found = 0;
        arbitages_found = 0;
        points_diff = 0;
        not_arbitage = 0;

        i = 0
        while(i < len(sportsbet_names)) :
            j = 0
            while(j < len(bet365_names)) :
                if (sportsbet_names[i] == bet365_names[j]) :
                    games_found+=1

                    if(sportsbet_points[i] != bet365_points[j]):
                        points_diff+=1
                        j+=1
                        continue

                    if sportsbet_overOdds[i] > bet365_overOdds[j] :
                        best_over = sportsbet_overOdds[i]
                        best_under = bet365_underOdds[j]
                        overBest = "SportsBet"
                        underBest = "Bet365"
                    else:
                        best_over = bet365_overOdds[j]
                        best_under = sportsbet_underOdds[i]
                        overBest = "Bet365"
                        underBest = "SportsBet"

                    # Arbitage formula -> https://youtu.be/TGinzvSDayU?t=333 
                    calc = (1/float(best_over) + 1/float(best_under)) * 100;

                    if (calc < 100) :
                        # print(game_name)
                        print("################################") 
                        print("ARBITAGE FOUND!!! -> " , sportsbet_names[i]) 
                        print("Profit margin of -> " , round(100-calc,3))

                        # print("\nSportsbet Over " , sportsbet_overOdds[i]) 
                        # print("Sportsbet Under " , sportsbet_underOdds[i])

                        # print("\nBet365 Over " , bet365_overOdds[j])
                        # print("Bet365 Under " , bet365_underOdds[j])
                        print("\nWith budget of $"+str(budget))

                        # Unbiased formula - amountToBet = budget/(odd1/odd2 + 1) -> http://www.aussportsbetting.com/guide/sports-betting-arbitrage/
                        print("\nBest Over " + overBest, best_over,"-> $"+ str(round((budget/((float(best_over)/float(best_under))+1)),2)))
                        print("Best Under " + underBest, best_under,"-> $" + str(round((budget/((float(best_under)/float(best_over))+1)),2)))
                        
                        # Over Biased
                        # print("\nBest Over " + overBest, best_over,"-> $"+ str(round((budget-budget/float(best_under)),2)))
                        # print("Best Under " + underBest, best_under,"-> $" + str(round(budget/float(best_under),2)))

                        # Under Biased
                        # print("\nBest Over " + overBest, best_over,"-> $"+ str(round(budget/float(best_over),2)))
                        # print("Best Under " + underBest, best_under,"-> $" + str(round((budget-budget/float(best_over)),2)))
                        print("################################") 
                        arbitages_found+=1;
                    else:
                        not_arbitage+=1;
                j+=1
            i+=1

        print("\nPotential arbitages:" , games_found)
        print("Arbitages found:" , arbitages_found)
        print("Games with different points:" , points_diff)
        print("Not arbitages:" , not_arbitage)

    # def lets_go_json(self):
